File: atcoder/arc/arc092_b.py
import bisect
import sys
if sys.version[0] == "2":
    range, input = xrange, raw_input

N = int(input())
A = [int(x) for x in input().split()]
B = [int(x) for x in input().split()]

bits = [1 << i for i in range(31)]
ans = 0
for i in range(30):
    mod = bits[i + 1] - 1
    mod_A = [a & mod for a in A]
    mod_B = [b & mod for b in B]
    mod_A.sort()
    bit_count = 0
    for b in mod_B:
        bit_count += bisect.bisect_left(mod_A, bits[i + 1] - b) - bisect.bisect_left(mod_A, bits[i] - b)
        bit_count += N - bisect.bisect_left(mod_A, (bits[i + 1] | bits[i]) - b)
    if bit_count & 1:
        ans |= bits[i]
print(ans)

File: atcoder/arc/arc092_b.nim
# import sequtils, strutils, strscans, algorithm, math, future, sets, queues, tables # for yukicoder (0.17.1)
import sequtils, strutils, algorithm, math, future, sets, queues, tables # for AtCoder (0.13.0)
template getLine: string = stdin.readLine
template getInteger: int = getLine.parseInt
template getBiggestInteger: int64 = getLine.parseBiggestInt
template getIntSeq: seq[int] = getLine.split.map(parseInt)
template getBigIntSeq: seq[int64] = getLine.split.map(parseBiggestInt)

when isMainModule:
  let
    N = getInteger
    A = getIntSeq
    B = getIntSeq
  var ans = 0
  var MOD = newSeq[int]()
  for i in 0..30:
    MOD.add(1 shl i)

  for i in 0..28:
    var mod_A, mod_B = newSeq[int]()
    for a in A:
      mod_A.add(a and (MOD[i + 1] - 1))
    for b in B:
      mod_B.add(b and (MOD[i + 1] - 1))
    mod_B.sort(cmp)

    var cnt = 0
    for a in mod_A:
      cnt += mod_B.lowerBound(MOD[i + 1] - a) - mod_B.lowerBound(MOD[i] - a)
      cnt += mod_B.lowerBound(MOD[i + 2] - a) - mod_B.lowerBound(MOD[i + 1] + MOD[i] - a)
    echo(cnt)
    ans = ans or (MOD[i] * (cnt and 1))
  ans.echo
